src/clr_bundle_add.c

```diff
@@ -53,12 +53,15 @@ static void print_help(void)
 
 static const struct option prog_opts[] = {
 	{ "list", no_argument, 0, 'l' },
-	{ "skip-diskspace-check", no_argument, &skip_diskspace_check, 1 },
+	{ "skip-diskspace-check", no_argument, 0, 's' },
```

Comment:
yeah I don't love it either but I couldn't find another way to handle those long opts from the config file without having an equivalent short opt. The fact that we have a short option for every flag doesn't mean we need to publish/show that short option to the end user. So the end user won't know there is a short option for that flag. So this approach should not become a problem unless we run out of letters for the short options which I don't think is gonna happen, and if it does, we can change this approach then. Or what suggestions do you have for solving this?

Reply:
One of the things that make this problem hard to solve in a different way is that some options from the command line don't receive an argument, for example `--debug` which only sets the log level at "debug level". However, when using the same value from the config file the same flag will always have an argument, because we could enable a global flag for all commands but we may want to disable it only for one command, so we need to be able to pass a true/false to that flag that would take no argument otherwise.

Reply:
> The fact that we have a short option for every flag doesn't mean we need to publish/show that short option to the end user. 

yes, but the user could use that. And this limits us in the number of parameters that we may have. And it's getting hard to find new short options that makes sense.

> One of the things that make this problem hard to solve in a different way is that some options from the command line don't receive an argument, for example --debug which only sets the log level at "debug level". 

Another option is to create another function that is called by both parse_options and the config parser to set the values instead of making the config parser to call the parse_options.

Not sure if it helps you, but we could use the longindex parameter from getopt_long function to handle long parameters in a switch case too. Another approach I can think of is to set the val in struct options to a value that isn't a valid printed character and avoid adding that char to optstring. According to the manual that's only the value that will be returned by getopt_long() if the function is used. it doesn't affect the short option.